aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1360,19 +1369,245 @@ Tensor nuclear_norm(const Tensor& self, IntArrayRef dim, bool keepdim) {
 
 Tensor& nuclear_norm_out(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim) {
   TORCH_CHECK(dim.size() == 2, "nuclear norm requires a 'dim' argument of size 2");
+  auto dim_ = dim.vec();
+  maybe_wrap_dims(dim_, self.dim());
 
-  auto permutation = create_dim_backshift_permutation(dim[0], dim[1], self.dim());
+  auto permutation = create_dim_backshift_permutation(dim_[0], dim_[1], self.dim());
   auto permutation_reverse = create_reverse_permutation(permutation);
 
   Tensor p = self.permute(permutation);
   at::sum_out(result, std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/false)), -1, keepdim);
   if (keepdim) {
     result.unsqueeze_(-1);
-    result = result.permute(permutation_reverse);
+    Tensor result_ = result.permute(permutation_reverse);
+    at::native::resize_output(result, result_.sizes());
+    result.copy_(result_);
+  }
+  return result;
+}
+
+static std::vector<int64_t> make_dim_list(int64_t ndim) {
+  std::vector<int64_t> dim_list(ndim);
+  for (int64_t ind = 0; ind < ndim; ind++) {
+    dim_list[ind] = ind;
+  }
+  return dim_list;
+}
+
+static void check_str_ord_valid(std::string str_ord, optional<IntArrayRef> opt_dim, int64_t ndim, optional<ScalarType> opt_dtype) {
+  TORCH_CHECK((str_ord == "nuc") || (str_ord == "fro"), "Invalid norm order: ", str_ord);
+  TORCH_CHECK(!opt_dtype.has_value(), "dtype argument is currently not supported in frobenius norm, ",
```

Comment:
frobenius norm does not yet support the dtype argument

Reply:
Can you elaborate on this check? The previous check validates that the norm is either nuclear or frobenius, but then this dtype check references frobeniusn norm without checking that `str_ord` is `fro`? Could this trigger on nunclear norm calls, too?

Reply:
Yeah, the check is actually supposed to apply to both frobenius and nuclear. I'll add the order value to the error message.